01_fib/undrewb/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func sign(n int) int {
+	var result = 1
+	if n < 0 {
+		result = int(math.Pow(-1, float64(n+1)))
+	}
+	return result
+}
+
+func fib(n int) {
+	if n > 20 || n < -20 {
+		fmt.Fprintln(out, "I dont go past twenty")
+		return
+	}
+
+	var lastIdx = int(math.Abs(float64(n)))
+	for idx := 0; idx <= lastIdx; idx++ {
```

Comment:
A top-level `if n >= 0` around two for-loops would yield a cleaner code structure.

Reply:
Done